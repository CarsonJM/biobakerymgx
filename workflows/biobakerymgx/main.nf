/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    IMPORT LOCAL MODULES/SUBWORKFLOWS
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/


//
// SUBWORKFLOWS: Local subworkflows
//
include { FASTQ_READ_PREPROCESSING_KNEADDATA } from '../../subworkflows/kneaddata/main'
// TODO: Add metaphlan/download module
// TODO: Add metaphlan/metaphlan module
// TODO: Add metaphlan/mergetables module


/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    IMPORT NF-CORE MODULES/SUBWORKFLOWS
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

//
// MODULE: Installed directly from nf-core/modules
//
include { CAT_FASTQ } from '../../modules/nf-core/modules/cat/fastq/main'


/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    RUN MAIN WORKFLOW
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

// Info required for completion email and summary
def multiqc_report = []

workflow BIOBAKERYMGX {

    take:
    fastq_gz    // [ [ meta ], [ read_1.fastq.gz, read_2.fastq.gz ] , paired-end reads (mandatory)

    main:
    ch_versions = Channel.empty()


    /*-----------------------------------------------------------------------------------
        Read-preprocessing: KneadData
    -----------------------------------------------------------------------------------*/
    //
    // SUBWORKFLOW: KneadData
    //
    if ( params.run_kneaddata ) {
        ch_preprocessed_fastq_gz = KNEADDATA ( ch_merged_fastq_gz ).preprocessed_fastq_gz
        ch_versions = ch_versions.mix(KNEADDATA.out.versions)
    } else {
        ch_preprocessed_fastq_gz = ch_merged_fastq_gz
    }


    /*-----------------------------------------------------------------------------------
        Taxonomic classification: MetaPhlAn
    -----------------------------------------------------------------------------------*/
    //
    // SUBWORKFLOW: MetaPhlAn
    //
    if ( params.run_metaphlan ) {
        ch_preprocessed_fastq_gz = KNEADDATA ( ch_merged_fastq_gz ).preprocessed_fastq_gz
        ch_versions = ch_versions.mix(KNEADDATA.out.versions)
    } else {
        ch_preprocessed_fastq_gz = ch_merged_fastq_gz
    }

    emit:
    preprocessed_reads_fastq_gz = ch_preprocessed_fastq_gz
    versions                    = ch_versions
}

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    THE END
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/
