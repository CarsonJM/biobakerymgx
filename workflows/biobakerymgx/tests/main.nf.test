nextflow_workflow {

    name "Test workflow: BIOBAKERYMGX"
    script "workflows/biobakerymgx/main.nf"
    workflow "BIOBAKERYMGX"
    tag "workflows"
    tag "biobakerymgx"
    tag "biobakerymgx_default"

    test("") {
        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        [
                            file("$projectDir/assets/test-datasets/fastqs/SRX5235702_SRR8427717_1.fastq.gz", checkIfExists: true),
                            file("$projectDir/assets/test-datasets/fastqs/SRX5235702_SRR8427717_2.fastq.gz", checkIfExists: true),
                        ]
                    ],
                    [
                        [ id:'test2' ],
                        [
                            file("$projectDir/assets/test-datasets/fastqs/SRX5235702_SRR8427717_1.fastq.gz", checkIfExists: true),
                            file("$projectDir/assets/test-datasets/fastqs/SRX5235702_SRR8427717_2.fastq.gz", checkIfExists: true),
                        ]
                    ]
                )
                """
            }
            params {
                outdir                  = "$outputDir"
                kneaddata_db_version    = 'human_genome'
                metaphlan_db_version    = 'mpa_vJan21_TOY_CHOCOPhlAnSGB_202103'
                max_cpus                = 1
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
