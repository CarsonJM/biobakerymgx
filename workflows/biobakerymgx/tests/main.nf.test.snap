{
    "fastq.gz": {
        "content": [
            [
                [
                    {
                        "id": "test2"
                    },
                    [
                        "SRX5235702_SRR8427717_1.fastq.gz:md5,cc895d36dc655210366086535f359fb0",
                        "SRX5235702_SRR8427717_2.fastq.gz:md5,2b1a67a42fb71032d6c54776fad5a790"
                    ]
                ],
                [
                    {
                        "id": "test"
                    },
                    [
                        "SRX5235702_SRR8427717_1.fastq.gz:md5,cc895d36dc655210366086535f359fb0",
                        "SRX5235702_SRR8427717_2.fastq.gz:md5,2b1a67a42fb71032d6c54776fad5a790"
                    ]
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-24T22:51:47.767796873"
    }
}